python/lsst/daf/butler/formatters/fitsExposureFormatter.py

```diff
@@ -109,11 +89,19 @@ def readInfoComponent(self, component):
         obj : component-dependent
             In-memory component object.
         """
-        from lsst.afw.image import LOCAL
-        from lsst.geom import Box2I, Point2I
-        parameters = dict(bbox=Box2I(minimum=Point2I(0, 0), maximum=Point2I(0, 0)), origin=LOCAL)
-        tiny = self.readFull(parameters)
-        return self.fileDescriptor.storageClass.assembler().getComponent(tiny, component)
+        reader = ExposureFitsReader(self.fileDescriptor.location.path)
+        componentMap = {'wcs': 'readWcs',
+                        'coaddInputs': 'readCoaddInputs',
+                        'psf': 'readPsf',
+                        'image': 'readImage',
+                        'mask': 'readMask',
+                        'variance': 'readVariance',
+                        }
+        method = componentMap.get(component, None)
+        if method:
+            caller = getattr(reader, method, None)
+        if caller:
```

Comment:
I think this needs to be indented since if method is false caller will never be defined.